File: snippets/csharp/basic-xml-serialization_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create a simple text message
var message = new UserMessage
{
    MessageId = "msg-001",
    Contents = new List<AIContent>
    {
        new TextContent { Text = "Hello, how can you help me today?" }
    }
};

// Serialize to XML
var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);

Console.WriteLine(xmlOutput);

File: snippets/python/basic-xml-serialization_main.py
# from microsoft.agents.xml import MessageSerializer
# from microsoft.agents.xml.models import ChatMessage, TextContent
#
# # Create a simple text message
# message = ChatMessage(
#     role="user",
#     message_id="msg-001",
#     contents=[
#         TextContent(text="Hello, how can you help me today?")
#     ]
# )
#
# # Serialize to XML
# serializer = MessageSerializer()
# xml_output = serializer.serialize(message)
#
# print(xml_output)